docs: Add  descriptions of node types (#897)
oasisprotocol/docs: docs/node/run-your-node/README.mdx

to participate in the consensus process. This process involves validating
transactions, signing blocks, and ensuring the integrity of the blockchain.

[Validator Node]: ./validator-node.mdx

## Compute Nodes
[Compute Nodes] are responsible for executing smart contracts and processing
transactions within a specific ParaTime (Parallel Runtime). These nodes handle
the actual computation tasks, such as running decentralized applications
(dApps), performing data processing, and executing privacy-preserving smart
contracts.

- **Sapphire Compute Node** is responsible for executing EVM-compatible
privacy-preserving smart contracts and processing transactions within the
Sapphire ParaTime. These nodes validate and execute transactions while
maintaining the confidentiality of sensitive data, which is a crucial aspect
of applications that handle private information or require enhanced security.
This is achieved through trusted execution environments (TEEs) that ensure
data remains encrypted and confidential, even while being processed.

- **Cipher Compute Node** is responsible for executing privacy-preserving smart

> contracts and processing transactions within the Cipher ParaTime. These nodes

Reviewer: ```suggestion
contracts written in Oasis Wasm and processing transactions within the Cipher ParaTime. These nodes
```

validate and execute transactions while maintaining the confidentiality of
sensitive data, which is a crucial aspect of applications that handle private
information or require enhanced security. This is achieved through trusted
execution environments (TEEs) that ensure data remains encrypted and
confidential, even while being processed.

- **Emerald Compute Node** is responsible for executing EVM-compatible smart
contracts and processing transactions within the Emerald ParaTime. It performs
tasks such as validating transactions, running EVM-based smart contracts, and
ensuring that the operations within the Emerald ParaTime are carried out
efficiently.

[Compute Nodes]: ./paratime-node.mdx

## Client Nodes
A Client Node is a type of node within the Oasis Network that serves as
an interface for users or other applications to interact with the blockchain.
Unlike compute nodes, which handle transaction processing and smart contract
execution, client nodes are primarily responsible for tasks such as querying

> the blockchain, submitting transactions, and retrieving data from the network.

Reviewer: ```suggestion
the blockchain, submitting transactions, and retrieving other data from the network.
```

Network that interacts with the Sapphire ParaTime. The Sapphire ParaTime is
designed to support EVM-compatible confidential smart contracts and
privacy-preserving decentralized applications (dApps) with strong privacy
features and high performance.

- **[Cipher Client Node]** is a type of node within the Oasis Network designed
to interact specifically with the Cipher ParaTime. The Cipher ParaTime is known
for its strong privacy features, allowing for the execution of confidential
smart contracts and the development of privacy-preserving decentralized
applications (dApps).

- **[Emerald Client Node]** is a specific type of client node within the Oasis
Network, designed to interact with the Emerald ParaTime. The Emerald ParaTime is
an Ethereum-compatible environment on the Oasis Network, allowing developers
to deploy and manage decentralized applications (dApps) that utilize the
Ethereum Virtual Machine (EVM).

[Non-Validator Node]: ./non-validator-node.mdx
[Sapphire Client Node]: ./paratime-client-node.mdx
[Cipher Client Node]: ./paratime-client-node.mdx
[Emerald Client Node]: ./paratime-client-node.mdx

## Rosetta Gateway

> A Rosetta Gateway service is a specialized service within the Oasis

Reviewer: ```suggestion
A [Rosetta Gateway] service is a specialized service within the Oasis
```

> Network that implements the Rosetta API to provide a standardized and simplified

Reviewer: ```suggestion
Network that implements the [Rosetta API] to provide a standardized and simplified
```

interface for interacting with the blockchain. This service is crucial for
enabling seamless integration between the Oasis Network and various external
platforms, such as exchanges, wallets, custodians, and blockchain-based

> applications.

Reviewer: ```suggestion
applications. The Rosetta Gateway connects to a [non-validator](#client-nodes) node.
```

> 

Reviewer: ```suggestion

[Rosetta Gateway]: https://github.com/oasisprotocol/oasis-rosetta-gateway
[Rosetta API]: https://docs.cdp.coinbase.com/mesh/docs/api-reference
```

## Public gRPC

> An Oasis [Public gRPC] is a public API endpoint that allows developers to interact
> with the Oasis Network using the gRPC protocol, enabling efficient, real-time,
> and cross-platform communication with the blockchain.
> 
> [Public gRPC]: ../grpc.mdx

Reviewer: ```suggestion
The [Oasis gRPC protocol] is an efficient, real-time, and cross-platform communication
protocol that allows developers to both manage the Oasis node and communicate
with the Oasis network. While each Oasis node opens a gRPC socket, the public
endpoints run a proxy and expose only a small, safe subset of the API calls publicly.

[Oasis gRPC protocol]: ../grpc.mdx
```

## Web3 Gateway
A [Web3 Gateway] enables interaction with the Oasis Network using standard

> Web3 protocols, which are widely used in the Ethereum ecosystem. It acts as a

Reviewer: ```suggestion
Web3 protocol, which is widely used in the Ethereum ecosystem. It acts as a
```

bridge between Web3-based applications and the Oasis Network, allowing
developers to leverage the tools, libraries, and practices familiar in
Ethereum development while benefiting from the unique features of the Oasis

> Network, such as privacy and confidentiality.

Reviewer: ```suggestion
Network, such as privacy and confidentiality. The Web3 gateway connects to a [compute](#compute-nodes) or a [client](#client-nodes) Sapphire or Emerald node.
```

[Web3 Gateway]: ../web3.mdx

## Archive Node
An [Archive Node] is a specialized node within the Oasis Network that stores the
entire blockchain history, making it a crucial tool for in-depth analysis,
development, and ensuring that the network's past states remain accessible.

[Archive Node]: ./archive-node.md

## Seed Node
A [Seed Node] is a type of node in the Oasis Network that serves a critical role

> in helping other nodes discover peers and join the network. Unlike validator

Reviewer: ```suggestion
in helping other nodes discover peers and join the network (*address book*). Unlike validator
```

nodes, which participate in the consensus process, Seed Nodes do not play a
direct role in consensus.

[Seed Node]: ./seed-node.md

## Key Manager Node

> As [Key Manager Node] is a specialized node in the Oasis Network responsible for

Reviewer: ```suggestion
A [Key Manager Node] is a specialized node in the Oasis Network responsible for
```

securely managing cryptographic keys used in confidential computing. These nodes
are crucial for the network's privacy-preserving features, enabling secure
encryption and decryption of data processed within Trusted Execution
Environments (TEEs). They play a vital role in enabling the Oasis Network's
secure, decentralized, and privacy-focused operations.

[Key Manager Node]: ./keymanager-node/README.md

<DocCardList items={[
    findSidebarItem('/node/run-your-node/prerequisites'),
    findSidebarItem('/node/run-your-node/validator-node'),
    findSidebarItem('/node/run-your-node/paratime-node'),
    findSidebarItem('/node/run-your-node/non-validator-node'),
    findSidebarItem('/node/run-your-node/paratime-client-node'),
    findSidebarItem('/node/web3'),

> ]} />

Reviewer: The doc cards at the end of the document is usually meant for "See also" section. But the cards you provided are already mentioned in the document, so I would use it for sth else.

import {findSidebarItem} from '@site/src/sidebarUtils';
import DocCardList from '@theme/DocCardList';

> # Run your node

Reviewer: Please add new lines after the section titles.

```suggestion
# Run your node

```

applications. The Rosetta Gateway connects to a [non-validator](#client-nodes)
node.

[Rosetta Gateway]: https://github.com/oasisprotocol/oasis-rosetta-gateway
[Rosetta API]: https://docs.cdp.coinbase.com/mesh/docs/api-reference

### Public gRPC
The [Oasis gRPC protocol] is an efficient, real-time, and cross-platform
communication protocol that allows developers to both manage the Oasis node and communicate with the Oasis network. While each Oasis node opens a gRPC socket,
the public endpoints run a proxy and expose only a small, safe subset of the
API calls publicly.

[Oasis gRPC protocol]: ../grpc.mdx

### Web3 Gateway
A [Web3 Gateway] enables interaction with the Oasis Network using standard
Web3 protocol, which is widely used in the Ethereum ecosystem. It acts as a
bridge between Web3-based applications and the Oasis Network, allowing
developers to leverage the tools, libraries, and practices familiar in
Ethereum development while benefiting from the unique features of the Oasis
Network, such as privacy and confidentiality.  The Web3 gateway connects to a
[compute](#compute-nodes) or a [client](#client-nodes) Sapphire or Emerald node.

[Web3 Gateway]: ../web3.mdx

> <DocCardList items={[
>     findSidebarItem('/node/run-your-node/prerequisites'),
>     findSidebarItem('/node/run-your-node/troubleshooting'),
>     findSidebarItem('/node/run-your-node/advanced/copy-state-from-one-node-to-the-other'),
>     findSidebarItem('/node/run-your-node/advanced/sync-node-using-state-sync'),
> ]} />

Reviewer: ```suggestion
## See also

<DocCardList items={[
    findSidebarItem('/node/run-your-node/maintenance'),
    findSidebarItem('/node/run-your-node/advanced'),
    findSidebarItem('/node/run-your-node/troubleshooting'),
]} />
```
